### DISCUSSION TITLE
External Data Option 3 and Option 4

### DISCUSSION THREAD

**Discussion by Ronnie-personal** [open-policy-agent/community#376] [OPA and Rego]
Hi There,

I am currently reading the documentation on Open Policy Agent's external data options, specifically option 4 push data and option 3 bundle API, which can be found at https://www.openpolicyagent.org/docs/latest/external-data/#option-4-push-data and https://www.openpolicyagent.org/docs/latest/external-data/#option-3-bundle-api, respectively. My question is whether it is possible to include a static data file in an OPA bundle while also adding other JSON data files to the same OPA server using 'option 4 push data'.

Thanks!

**Comment by orweis** [comment]
Yes, you can.

Here's an article that dives more into these options covering more pros / cons :
https://www.permit.io/blog/load-external-data-into-opa

**Reply by Ronnie-personal** [reply]
Thanks for the prompt reply! I have reviewed the document, however it's not clear how it works when use both option 3 and 4 simultaneously on the same bundle.

**Reply by orweis** [reply]
The data you push on top, simply overrides the values that have been loaded before.
It's up to you to use the right PUT/PATCH calls to merge the data in the way you need.
